app/stores/NetworkStatusStore.js

```diff
@@ -199,6 +231,19 @@ export default class NetworkStatusStore extends Store {
     }
   };
 
+  _redirectToSyncingWhenLocalTimeDifferent = () => {
+    if (
+      this.localTimeDifference > this.ALLOWED_TIME_DIFFERENCE &&
+      !this.isSynced &&
+      this.stores.app.currentRoute !== ROUTES.PROFILE.LANGUAGE_SELECTION &&
+      this.stores.app.currentRoute !== ROUTES.PROFILE.TERMS_OF_USE &&
+      this.stores.app.currentRoute !== ROUTES.PROFILE.SEND_LOGS
```

Comment:
I think we should move these "if we are not on one of the setup pages" checks into a dedicated, reusable function. We have that already in a few places as far as i remember